packages/x-components/src/x-modules/related-prompts/events.types.ts

```diff
@@ -17,10 +18,20 @@ export interface RelatedPromptsXEvents {
    * The selected prompt has changed.
    * Payload: The index of the prompt in the RelatedPrompts list or -1 to remove selection.
    */
-  UserSelectedARelatedPrompt: number;
+  UserSelectedARelatedPrompt: { promptId: string; query: string };
   /**
    * The selected next query of the selected prompt has changed.
    * Payload: The index of the next query in the NextQueries list or -1 to remove selection.
    */
   UserSelectedARelatedPromptQuery: number;
+  /**
+   * A custom query to request has been provided.
+   * Payload: The query to add to the request.
+   */
+  RelatedPromptsCustomQueryProvider: string;
+  /**
+   * The current location of the related prompts.
+   * Payload: The query to request prompts and the location of the related-prompts.
+   */
+  RelatedPromptsLocation: { location: FeatureLocation | undefined; query: string };
```

Comment:
I think we don't need this event at all. We could just use UserSelectedARelatedPrompt and rely on the metada. We always provide the location as metadata for the events:

![image](https://github.com/user-attachments/assets/0f2a00de-e17c-4ef6-be0a-3a15b7a8927d)